BEIMA.Client/src/pages/Devices/AddDevicePage.js

```diff
@@ -19,28 +20,38 @@ const AddDevicePage = () => {
     "Notes": ""
   }
 
-  const [deviceFields, setDeviceFields] = useState(defaultDeviceFields);
+  const [deviceFields, setDeviceFields] = useState(currentDeviceFields);
+  const [errors, setErrors] = useState(currentDeviceFields);
   const [setPageName] = useOutletContext();
   const [deviceImage, setDeviceImage] = useState();
   const [deviceAdditionalDocs, setAdditionalDocs] = useState();
   const [fullDeviceJSON, setFullDeviceJSON] = useState({});
-
+  
   useEffect(() => {
     setPageName('Add Device')
   })
-
+  
   // gathers all the input and puts it into JSON, files are just assigned to state variables for now
   function createJSON(addButtonEvent){
     let formFields = addButtonEvent.target.form.elements;
     let fieldValues = {};
+    let newErrors = {};
 
     for(let i = 0; i < formFields.length; i++){
       let formName = formFields[i].name;
       let fieldNames = Object.keys(deviceFields);
 
       if(fieldNames.includes(formName)){
         let formJSON =  {[formName] : formFields[i].value};
-        formFields[i].value = "";
```

Comment:
Can we keep this line, but move it to a better spot? I use it to clear the fields on a submit. Or maybe we can create a separate method that clears the fields after all the input validation is done and passes

Reply:
I had tried to post a reply earlier but it doesn't look like it took.

Clearing the from will be in the Else statement when detecting errors. I'm hoping that setting the DB JSON object for the fields by selected device type to the currentDeviceFields variable will clear it. I've got some in progress stuff for that in another (unpublished at the moment) branch